lib/projections/ecto.ex

```diff
@@ -67,29 +66,39 @@ defmodule Commanded.Projections.Ecto do
 
         multi =
           Ecto.Multi.new()
-          |> Ecto.Multi.run(:verify_projection_version, fn repo, _changes ->
-            version =
-              case repo.get(ProjectionVersion, projection_name, prefix: prefix) do
-                nil ->
-                  repo.insert!(
-                    %ProjectionVersion{
-                      projection_name: projection_name,
-                      last_seen_event_number: 0
-                    },
-                    prefix: prefix
-                  )
-
-                version ->
-                  version
+          |> Ecto.Multi.insert_all(
+            :insert_version,
+            ProjectionVersion,
+            [
+              %{
+                inserted_at: DateTime.to_naive(DateTime.utc_now()),
+                updated_at: DateTime.to_naive(DateTime.utc_now()),
+                projection_name: projection_name,
+                last_seen_event_number: 0
+              }
+            ],
+            on_conflict: :nothing,
+            prefix: prefix
+          )
+          |> Ecto.Multi.update_all(
+            :update_version,
+            from(version in ProjectionVersion,
+              select: version,
+              where: version.projection_name == ^projection_name,
+              where: version.last_seen_event_number < ^event_number
+            ),
+            [set: [last_seen_event_number: event_number]],
+            prefix: prefix
+          )
```

Comment:
These two separate `INSERT` and `UPDATE` statements will need to be combined as a single [`UPSERT` query](https://hexdocs.pm/ecto/constraints-and-upserts.html#upserts) otherwise there is _still_ the possibilty of a concurrency issue due to executing two queries.